state/src/replicationv2/raft.rs

```diff
@@ -106,12 +109,8 @@ impl RaftClient {
 
         // Initialize the raft
         if config.init {
-            let mut initial_nodes = config.initial_nodes.clone();
-            if !initial_nodes.contains(&config.id) {
-                initial_nodes.push(config.id);
-            }
```

Comment:
why do we no longer do this? is the node expected to add itself to the cluster membership only after a round-trip of being discovered by another node in the gossip layer (triggering a proposal to add the local node as a learner)?

Reply:
Yes, it's expected a node will configure this value with its own node info. 

The reason this is messy at this layer is that the `initial_nodes` type is now `BTreeMap<NodeId, Node>` rather than `BTreeSet<NodeId>`. The ladder (and former implementation) fails to store the node info -- including the peer ID. This means that when membership is disseminated to new learners, they lose the information needed to dial the leader.

Instead, we need to construct a `BTreeMap` with the node info. This _could_ happen here, but it's cleaner to do at a higher layer where we have access to the relayer config and thereby the local peer's `PeerId`